services/MerchStoreService.ts

```diff
@@ -502,16 +642,24 @@ export default class MerchStoreService {
   }
 
   /**
-   * TODO: Rewrite deletePickupEvent as per spec: https://github.com/acmucsd/membership-portal/issues/204
+   * Deletes a pickup event. Before deletion, all orders for the pickup event will
+   * have emails sent out to the users who've placed the order.
    */
   public async deletePickupEvent(uuid: Uuid): Promise<void> {
     return this.transactions.readWrite(async (txn) => {
       const orderPickupEventRepository = Repositories.merchOrderPickupEvent(txn);
+      const orderRepository = Repositories.merchOrder(txn);
       const pickupEvent = await orderPickupEventRepository.findByUuid(uuid);
       if (pickupEvent.orders.length > 0) {
         throw new UserError('Cannot delete an order pickup event that has orders assigned to it');
       }
-      // Manually set all the orders' pickup events to null before deleting event
+      // email order cancellation for order, update order status, and set pickupEvent to null before deleting
+      pickupEvent.orders.forEach(async (order) => {
```

Comment:
If `forEach` is blocking then let's do these concurrently so we're not sending emails one at a time and blocking on each one.

Reply:
Looked at the email template to see if we could just send the same email to all the orderers, and it seems not, but it looks like we don't specify a URL here? Or am I misreading the HTML?

https://github.com/acmucsd/membership-portal/blob/OrderStates/templates/orderPickupCancelled.ejs#L120

Reply:
And can we send the same email to all the orderers? e.g. "Hey your order got canceled" and then instead of showing the order in the email just link to it?

Reply:
Good catch, I meant to ping Steven about what URL to use there but I never ended up doing that.

> And can we send the same email to all the orderers? e.g. "Hey your order got canceled" and check the portal to see which order? Kind of a dick move though, but it means we only send one email.

Haha interesting idea, I think it might be better UX if we send each person a separate email though so that they can see what specific items were in their cancelled order and so they each get their own URL to reschedule. Might need to look if there's some kind of rate limiting on Sendgrid's end to make sure mass mailing won't fail for some emails.